linter.py

```diff
@@ -1,57 +1,72 @@
 #
 # linter.py
-# Linter for SublimeLinter3, a code checking framework for Sublime Text 3
+# Linter for SublimeLinter4, a code checking framework for Sublime Text 3
 #
 # Written by Noah Davis
 # Copyright (c) 2016 Noah Davis
 #
+# Updated by Schuyler Jager & Andreas for SublimeLinter 4
+# Copyright (c) 2020 Schuyler Jager & Andreas
 # License: MIT
 #
 
 """This module exports the Codeclimate plugin class."""
+import logging
+import os
+from SublimeLinter.lint import Linter, util
 
-from SublimeLinter.lint import Linter, util, persist
+
+logger = logging.getLogger('SublimeLinter.plugin.eslint')
 
 
 class Codeclimate(Linter):
     """Provides an interface to codeclimate."""
-
-    syntax = (
-        'ruby on rails',
-        'ruby',
-        'javascript',
-        'python',
-        'go',
-        'html',
-        'css',
-        'markdown',
-        'php',
-        'haskall',
-        'java',
-        'shellscript'
-    )
-    executable = "codeclimate"
+    defaults = {
+        'selector': (
+            'source.css, '
+            'source.go, '
+            'source.haskell, '
+            'source.java, '
+            'source.javascript, '
+            'source.js, '
+            'source.php, '
+            'source.python, '
+            'source.ruby, '
+            'source.scss, '
+            'source.shell, '
+            'text.html'
+        )
+    }
     regex = r'^(?P<line>\d+)(?:-\d+)?:\s(?P<message>.+)$'
```

Comment:
If possible try to extract the filename the error is coming from to the group `(?P<filename>)`.  If that's possible SublimeLinter will handle errors from multiple files without any warnings.

Reply:
The output of codeclimate is:

```text
== src/Plugin/Field/FieldType/ACLFieldsItem.php (1 issue) ==
29: Define a constant instead of duplicating this literal "value" 3 times. [sonar-php]
```

I set `multiline` to `true` and [updated the regexp][1] to:

```text
regex = r'^== ((?P<filename>.*)(?= \(\d+ issue\) ==))( \(\d+ issue\) ==\n(?P<line>\d+))(?:-\d+)?:\s(?P<message>.+)$'
```

As far as I can see, it is working as expected. The doubled "\(\d+ issue\)" isn't that nice, but I found no other way to exclude the text from the file path with a positive lookahead and then set the second start in the same position for the rest.

[1]:https://github.com/codeclimate/SublimeLinter-contrib-codeclimate/pull/3/files#diff-e7c2c765d5c484aae3f449e93444a2659ddc4bac3b83d10c6ee5c36f6c9e8eebR38

Reply:
The file name here looks like a header; I doubt codeclimate will duplicate the filename information; so this won't extract multiple issues from one file I think.  Have you tested this?

Reply:
Oh… Oh! Yes, it is. Thank you for your close inspection. I reverted the regex in https://github.com/codeclimate/SublimeLinter-contrib-codeclimate/pull/3/commits/b7fa65184f8aa1de1087d5d6c6667efe1686b3cd.

Reply:
@kaste, I suggest preparing [my repository](https://github.com/meengit/SublimeLinter-contrib-codeclimate) to file the PR to overtake this linter plugin's maintenance in the next days. I will let the `master` branch, pointing to this pull request, as is, and switch with the *protected branch* to a newly created branch called `main` for further development.